pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -403,49 +403,48 @@ private void ensureMinReplicas(TableConfig config) {
   /**
    * Verify table configs if it's a hybrid table, i.e. having both offline and real-time sub-tables.
    */
-  private void verifyTableConfigs(TableConfig tableConfig) {
-    String rawTableName = TableNameBuilder.extractRawTableName(tableConfig.getTableName());
-    TableConfig offlineTableConfig = null;
-    TableConfig realtimeTableConfig = null;
-    if (tableConfig.getTableType() == CommonConstants.Helix.TableType.REALTIME) {
-      realtimeTableConfig = tableConfig;
+  private void verifyTableConfigs(TableConfig newTableConfig) {
+    String rawTableName = TableNameBuilder.extractRawTableName(newTableConfig.getTableName());
+    LOGGER.info("Validating table configs for Table: {}", rawTableName);
+
+    TableConfig tableConfigToCompare = null;
+    if (newTableConfig.getTableType() == CommonConstants.Helix.TableType.REALTIME) {
       if (_pinotHelixResourceManager.hasOfflineTable(rawTableName)) {
-        offlineTableConfig = _pinotHelixResourceManager.getOfflineTableConfig(rawTableName);
+        tableConfigToCompare = _pinotHelixResourceManager.getOfflineTableConfig(rawTableName);
       }
     } else {
-      offlineTableConfig = tableConfig;
       if (_pinotHelixResourceManager.hasRealtimeTable(rawTableName)) {
-        realtimeTableConfig = _pinotHelixResourceManager.getRealtimeTableConfig(rawTableName);
+        tableConfigToCompare = _pinotHelixResourceManager.getRealtimeTableConfig(rawTableName);
       }
     }
 
     // Check if it is a hybrid table or not. If not, there's no need to verify anything.
-    if (offlineTableConfig == null || realtimeTableConfig == null) {
+    if (tableConfigToCompare == null) {
+      LOGGER.info("Table: {} is not a hybrid table. Skip validating table configs.", rawTableName);
```

Comment:
The log message is misleading. We are not skipping validation of table configs. We don't have anything to check for consistency. I suggest making the log message as "Skipping consistency check across realtime and offline parts of the table" or you can just remove the log message altogether.

Reply:
Great! Thanks for the suggestion. I think it's good to keep this message log, since this message wouldn't be printed quite frequent, and it's beneficial for us to know how the request goes. I've adjusted the message in the latest push.